### DISCUSSION TITLE
Valhalla export routes - waypoints

### DISCUSSION THREAD

**Discussion by wavard** [devemux86/cruiser#66] [Q&A] [answered]
Is there a way to import routes without additional waypoints being added. At the moment at least two additional waypoints are added.

**Comment by devemux86** [comment]
Hi @wavard!

What kind of routes, GPX files?
Can you explain with more details with a route example to test?

You can select in the import dialog which GPX item will be used in the routing:
[Automatic - Track - Route - Waypoints]

The "Route" / "Waypoints" options use these GPX items exactly as they are in the gpx file.
The "Track" option does a sampling on the track points using the "Max waypoints" option.

**Reply by 0709wiwiwi** [reply]
@wavard 
If you only need to display  a '**naked**' gpx track use the 'hamburger' menu:
Tools > GPS recording > Import GPX track.
Set your the prefered track appearance by the same 'hamburger' menu:
Settings > GPS recording > Appearance >- Track color & Track width & Track style

**Reply by devemux86** [reply]
> Tools > GPS recording > Import GPX track

This imports GPX tracks into the "GPS recording" database (changing your recordings).

To overlay GPX tracks it is better to use the menu "Routing | Import" and select "Overlay".

**Comment by 0709wiwiwi** [comment]
Correct but the appearance menu choice by overlay (to show thé naked track only) is limited.

**Comment by wavard** [comment]
Ok. This is the situation. 

I generate the route in Cruiser. I export the route in gpx format for later use in Cruiser. I import the GPX route into Cruiser (set options to Routing & Max wypoints 2). When  investigating the Waypoints I notice there are additional Waypoints. These additional Waypoints are located 0m from Waypoints that I had set.

Now I notice that if I export the route in Kurviger format, the route is imported into Cruiser as I had intended ie no additional Waypoints at 0m from another Waypoint. 

My observation, otherwise a great app.

**Reply by devemux86** [reply]
Cruiser can export routes for later use in 2 **offline** formats:
(without need for new route calculation in import)
- GPX navigation track (select "**Navigation**" in export)
- Kurviger route

These routes can then be re-imported offline without changes and new calculation.

Do you select "Navigation" in export dialog?
Can you attach a route example to check?

**Reply by 0709wiwiwi** [reply]
Notice 
1. At the Gpx export > select Navigation. (Cruiser exports so a navigation track )
2. And at the Gpx (re)import > select **Automatic** or track (Cruiser will recognise the import is a navigation track)

**Reply by MotoUKRider** [reply]
I couldn't replicate this when I created a 2-waypoint route of around 40miles.  You *do* see 2 items in a popup box if your route travels over (and includes) an existing Favorite.  The popup will show an orange star and a numbered item icon.  Quite normal.

What did make me write this post, is that I tried editing the saved route and save the file a few minutes later - it saved a duplicate filename that couldn't be opened (presumably because Cruiser can't handle the naming?):

20230303_094244.gpx
20230303_094244.gpx (1)

By deleting the original file and removing the " (1)" using my onboard File Manager, it loads fine.  It doesn't look like the DateTime function is updating the auto-filled "Save" dialog.

**Reply by devemux86** [reply]
> At the Gpx export > select Navigation. (Cruiser exports so a navigation track )

:+1: 

> And at the Gpx (re)import > select Automatic or track (Cruiser will recognise the import is a navigation track)

No need for this! :slightly_smiling_face: 
Cruiser automatically detects the GPX navigation tracks and imports them.

(at least it will work that way in the next version)

**Reply by devemux86** [reply]
> 20230303_094244.gpx (1)

Android unfortunately changes the file extension(!) instead of the file name when duplicating files.

When exporting files and want to overwrite them:
- _Click on an existing file_
- Then press "Save" button
- An Android dialog appears to ask for overwrite

<img src="https://user-images.githubusercontent.com/3484020/222692545-3c29f571-be76-45ac-b1ed-09bb1ab572f8.jpg" width="50%"/>

**Reply by 0709wiwiwi** [reply]
> Aha roger I notice "**in the next version**" ;-) 
The current import is like this. 
Failure if before you had selected import route.

**Reply by MotoUKRider** [reply]
> > 20230303_094244.gpx (1)
> Android unfortunately changes the file extension(!) instead of the file name when duplicating files.

I think there's a problem here - the filename text remains the same, so if I press "Save" it should intercede with that dialog box asking if I want to overwrite rather than having me select a file from the list.

My complaint here is that the process I have to follow isn't "obvious".  I've got a filename in the edit box, but it's doing its own thing without me knowing.  Is Cruiser able to "handle" this better?

**Reply by devemux86** [reply]
> Is Cruiser able to "handle" this better?

This is an Android dialog, the apps can not change how it works.
This is provided by the operating system, having its own workflow.

However, I will check again to see if anything has changed. :slightly_smiling_face:

**Reply by wavard** [reply]
> Cruiser can export routes for later use in 2 **offline** formats: (without need for new route calculation in import)
> 
> * GPX navigation track (select "**Navigation**" in export)
> * Kurviger route
[test.zip](https://github.com/devemux86/cruiser/files/10881278/test.zip)

> 
> These routes can then be re-imported offline without changes and new calculation.
> 
> Do you select "Navigation" in export dialog? Can you attach a route example to check?

Attached is a sample route you requested

**Reply by 0709wiwiwi** [reply]
No attachment.  
To upload a file in github add .txt after the .gpx extension.

**Reply by wavard** [reply]
Now the file should be attached. 
[test.zip](https://github.com/devemux86/cruiser/files/10881300/test.zip)

**Reply by devemux86** [reply]
@wavard thanks for the sample!

I can reproduce the issue, it happens at export with Valhalla routing.
The export with other routing services seems to work properly.

I will check what happens.

**Reply by 0709wiwiwi** [reply]
By the gpx editor program I inspected the file. No idea how you produced that result. 
EDIT. _(Okido I only now did see  the above discovery by Emux when so using the Valhalla router.)_
I removed the first via and kept the second via with the streetname in the gpx trkpt <cmt> .
I than imported this so corrected file into Cruiser, and again exported as a new gpx file.
This new gpx file export does not contain any such double via on the exact same location.

[Export_Cruiser_wavard_test.gpx.txt](https://github.com/devemux86/cruiser/files/10881916/Export_Cruiser_wavard_test.gpx.txt)

The gpx editor export report.
Opening Export_Cruiser_wavard_test.gpx
0 waypoints
0 routes (0 points))
1 tracks (1180 points in 1 track segments)
TGPX
Track Segment #1 (1180 points)
 160.9 km / 3:34:13 = 45.0 km/h

**Comment by devemux86** [comment]
The issue was in Cruiser's parser of Valhalla routes (visible in export).

It will work properly in the next Cruiser update.

**Reply by 0709wiwiwi** [reply]
Roger, Emux, 

> The issue was in Cruiser's parser of Valhalla routes (**visible in export**).

Indeed  :-))

![image](https://user-images.githubusercontent.com/54167201/222742552-50282b14-54b7-434a-8f01-21fc7626230e.png)

**Comment by devemux86** [comment, marked as answer]
It should work properly in [Cruiser 3.0.11](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by wavard** [comment]
Thanks
Will look out for it
